src/promptflow-core/promptflow/executor/_tool_resolver.py

```diff
@@ -222,33 +223,67 @@ def _load_script_tool_as_function(self, node_name: str, assistant_tool: Assistan
         # Step IV: construct the AssistantTool object from the updated predefined_inputs + Tool object
         return self._construct_assistant_tool(tool, callable, updated_predefined_inputs)
 
-    def _generate_tool_definition(self, func_name: str, description: str, predefined_inputs: list) -> dict:
+    def _generate_tool_definition(self, tool: Tool, func: Callable, predefined_inputs: dict) -> dict:
         try:
-            to_openai_type = {
-                "str": "string",
-                "int": "number",
-                "float": "number",
-                "bool": "boolean",
-                "list": "array",
-                "dict": "object",
+            sig = inspect.signature(func)
+            parameters = sig.parameters
+            # Attempt to extract the description, handling exceptions
+            try:
+                description, params = DocstringParser.parse_description(func.__doc__)
+            except Exception as e:
+                # Log the exception if necessary
+                logging.warning(f"Failed to parse docstring for function {func.__name__}: {e}")
+                # Set description to None or an empty string
+                description = None  # or description = ""
+                params = {}
+
+            func_definition = {
+                "name": func.__name__,
+                "description": description,
+                "parameters": {"type": "object", "properties": {}, "required": []},
             }
-            description, params = DocstringParser.parse(description)
-            for input in predefined_inputs:
-                if input in params:
-                    params.pop(input)
-            for _, param in params.items():
-                param["type"] = to_openai_type[param["type"]] if param["type"] in to_openai_type else param["type"]
+
+            for name, param in parameters.items():
```

Comment:
I remember we already have some function to parse a function to a tool definition, can we use that logic instead of writing another parsing logic?